test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataUtf8JsonWriterTests.cs

```diff
@@ -169,6 +169,24 @@ public void WritePrimitiveValueString()
             this.VerifyWritePrimitiveValue("string", "\"string\"");
         }
 
+        [Fact]
+        public void WritePrimitiveValueLargeString()
+        {
+            this.VerifyWritePrimitiveValue(
+                new string('x', 30000),
+                "\"" + new string('x', 30000) + "\""
+                );
+        }
+
+        [Fact]
+        public void WritePrimitiveValueLargeStringWithSpecialChars()
+        {
+            this.VerifyWritePrimitiveValue(
+                new string('x', 20000) + "!@#$%" + new string('x', 10000),

```

Comment:
Do these chars really need to be escaped? I think you could add chars that need to be escaped like `"` as well as non-ascii characters that need special utf-8 encoding.

Reply:
None of the characters in this string are "special chars" (they won't be escaped). See: https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRj1UwAJUCA6AFQFMAPYSgKQGcB7KAbj19wDcAhmFKtSAXlJRaAd3IFMACgDk9ZTFIKAlKUSkARAEJUqAMQASAKT7dU2fKWr1mzFp4DhokZLacAyrRgAJaCADZBAF6BlAHBYZG0iqxueEIiDAAOtADGwLRwEqTm+gA6+gDerAC+ZfruFACcSSn4BE2sYC2NisUAIkEFtKzZghlBUADmpOzZ2RBgIKSVIoaSmTl5cFX6LUA

So this test does not do what it intends (based on the name at least). Consider adding special chars in there (e.g. double quote, control characters like new lines, non ascii characters).

Reply:
Yes.I'll update it.